pkg/bitnet/internal/model/loader.go

```diff
@@ -0,0 +1,152 @@
+package model
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"io"
+	"io/fs"
+	"sync"
+)
+
+var (
+	ErrModelNotFound = errors.New("model file not found")
+	ErrInvalidGGUF   = errors.New("invalid GGUF magic number")
+	ErrModelNotSet   = errors.New("model path not set")
+	ErrReaderNil     = errors.New("reader is nil")
+)
+
+// GGUFHeader represents the header of a GGUF format file
+type GGUFHeader struct {
+	Magic       uint32
+	Version     uint32
+	TensorCount uint64
+	KVCount     uint64
+}
+
+// ModelLoader handles loading and managing the BitNet model file in GGUF format.
+type ModelLoader struct {
+	fs         fs.FS
+	modelPath  string
+	bufferSize int
+	chunkPool  sync.Pool
+	header     *GGUFHeader
+}
+
+// NewModelLoader creates a new ModelLoader instance.
+func NewModelLoader(filesystem fs.FS, modelPath string) (*ModelLoader, error) {
+	if filesystem == nil {
+		return nil, errors.New("filesystem cannot be nil")
```

Comment:
These errors should be written in the above `var ( ... )`. Whole point is so that we don't need to allocate.